index.css

```diff
@@ -1 +1,180 @@
+body
+{
+    display: flex;
+    justify-content: space-between;
+    flex-direction: column;
+}
 
+input[name='kinds']
+{
+    display: none;
+}
+
+#dog:checked ~ section .cat,
+#ufo:checked ~ section .cat,
+#hamster:checked ~ section .cat,
+#cat:checked ~ section .dog,
+#ufo:checked ~ section .dog,
+#hamster:checked ~ section .dog,
+#cat:checked ~ section .ufo,
+#dog:checked ~ section .ufo,
+#hamster:checked ~ section .ufo,
+#cat:checked ~ section .hamster,
+#dog:checked ~ section .hamster,
+#ufo:checked ~ section .hamster
+{
+    opacity: .5;
+}
```

Comment:
#dog:checked ~ section :not(.dog) { opacity: 0.5 }
#ufo:checked ~ section :not(.ufo) { opacity: 0.5 }
#cat:checked ~ section :not(.cat) { opacity: 0.5 }
#hamster:checked ~ section :not(.hamster) { opacity: 0.5 }